step/serialization.md

```diff
@@ -1,109 +1,553 @@
-# Serialization holder
+# Serialization
 
 ## Overview
 
-This 5-minute optional lab will show you how **`jlink`** can greatly improve the size of a Java runtime image.
+Serialization has long been a sticky subject in Java. Brian Goetz describing it as a paradox, a key to Java's success allowing for transparent remoting, but also the implementation of serialization making a number of key mistakes. 
 
-...
+A short list of mistakes include;
+* Bypasses constructors and encapsulation  
+* The behavior of serialization depends on "magic" methods (including `readObject()` which we will look at later)
+* Poor stream format that is not efficent, reusable, nor human readable 
 
-## Using `jlink`
+In this section we will explore some of the short comings with Serialization in Java, how some of these issues have been addressed in recent JEPs, and how Records addresses many of the issues with serialization. 
```

Comment:
I wouldn't mention Records here and put the whole Records part more as a side-note